### PR TITLE
tests: pass data dir as arg

### PR DESCRIPTION
**Pull request by radfish**
This fixes test failure on builds that happen
to be built in 'build/' instead of 'build/release'.

### CODE REVIEW

**Comment by radfish** [comment]
@moneromooo-monero You mean gtest's options? The gtest call eats all its option and leaves the rest. So, it shouldn't barf -- could you explain more?

Re /: I wanted to support empty string as the data path, to mean current working directory. So, it's checking if path is empty to avoid pointing to root, like "" -> "/datafile".

Re boost: yeah, ok, i'll look those up.

Re default run: cmake passes it the data dir arg now. If you mean a manual run -- it will cleanly error out if the user did not pass it the data dir, otherwise it will work You want it to work without passing any argument? Then we would hav to be back to "../../../" hacks, no? Could you explain what you have in mind under "default run"?

**Comment by moneromooo-monero** [comment]
I had missed the call to test init before the argv usage, that's fine then. By default run, I meant if you run just "./build/debug/tests/unit_tests/unit_tests". Why break it ? I suppose it's not too annoying to have to pass the pathe every time though, it just seems arbitrarily annoying :)

**Comment by radfish** [comment]
Brought back default with no args that assumes build/release.
Replaced string with boost path type.

**Comment by danrmiller** [comment]
This fixes segfault 11 I was getting on osx for Serialization.portability_wallet

File: tests/unit_tests/serialization.cpp [hunk 1]
```diff
@@ -671,12 +672,12 @@ TEST(Serialization, portability_wallet)
...
   const bool restricted = false;
   tools::wallet2 w(testnet, restricted);
-  string wallet_file = epee::string_tools::get_current_module_folder() + "/../../../../tests/data/wallet_9svHk1";
+  const boost::filesystem::path wallet_file = unit_test::data_dir / "wallet_9svHk1";
   string password = "test";
   bool r = false;
   try
   {
-    w.load(wallet_file, password);
+    w.load(wallet_file.native(), password);
```
Comment (moneromooo-monero) [inline tests/unit_tests/serialization.cpp]: Can you explain this please ? It looks like it ought to be a separate patch maybe,

Comment (radfish) [reply tests/unit_tests/serialization.cpp]: It's due to switching from string to boost::filesystem::path

**Comment by radfish** [comment]
@anonimal yeah, it should. Btw, you've probably seen this already, but you can disable only failing tests in the PKGBUILD: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=monero-git#n60

Btw, off topic: I've recently seen a package that *builds* tests in check(). We should try to do that too, so that `makepkg --nocheck` build becomes faster. Should be as easy as adding `cmake -DBUILD_TESTS=ON && make` in check(), but I haven't tried.

File: tests/unit_tests/serialization.cpp (hunk 1 above)
Comment (radfish) [reply tests/unit_tests/serialization.cpp]: We could keep wallet_file as a string and convert up above where wallet_file is declared, but the current way seems marginally better because boost::filesystem::path is the natural type for wallet_file not string. Splitting hairs.

Comment (moneromooo-monero) [reply tests/unit_tests/serialization.cpp]: OK, thanks.
